crawl-ref/source/webserver/static/scripts/client.js

```diff
@@ -1709,5 +1713,29 @@ function (exports, $, key_conversion, chat, comm) {
         return x > 0 ? x : -x;
     }
 
+    function is_mobile()
```

Comment:
Anything in `static` is not versioned, and so updating both at the same time usually leads to problems (stale caches, servers needing to restart). I'm not immediately sure what a clean solution is for this code (i.e. something that doesn't touch `client.js`), which does seem to need access minimally to `handle_keypress`. For making changes that do touch both versioned and non-versions js code it is usually best to write the versioned code so that it doesn't crash if the `client.js` hook is missing (and this can be removed later once servers catch up). The change here would probably be to make the `game.js` event handlers only installed if `is_mobile` is present in `client`. But if you can figure out some way to do this without touching `client.js` that might be better.

Reply:
I'll try to move as much code as possible to game.js. Is that right?

There are also changes in static/style.css. Should I use game_data/static/style.css instead?

Reply:
Done.

Reply:
(yes, css changes in `game_data` are also preferred unless it is something that is completely version-agnostic. But sometimes it has to happen.)